research/engine/pegasus/pegasus/src/operator/concise/limit.rs

```diff
@@ -60,3 +66,230 @@ impl<D: Data> Limit<D> for Stream<D> {
         })
     }
 }
+
+impl<D: Data + Ord> SortLimit<D> for Stream<D> {
+    fn sort_limit(self, size: u32) -> Result<Stream<D>, BuildJobError> {
+        self.sort_limit_by(size, |x, y| x.cmp(y))
+    }
+}
+
+impl<D: Data> SortLimitBy<D> for Stream<D> {
+    fn sort_limit_by<F>(self, size: u32, cmp: F) -> Result<Stream<D>, BuildJobError> where F: Fn(&D, &D) -> Ordering + Send + 'static {
+        let share_cmp = ShadeCmp { cmp: Rc::new(cmp) };
+        let cmp_clone = ShadeCmp { cmp: share_cmp.cmp.clone() };
+
+        let local_sort = sort_limit_by_partition(self, size, cmp_clone, false)?;
```

Comment:
Better define `sort_limit_by_partition(.., name: String, )`, and we can give the two calls different names.
One with `sort_limit_by_partition(..., "sort_limit_by_partition_locally")`
The other with `sort_limit_by_partition(..., "sort_limit_by_partition_globally")`